frontend/web/src/app/contest/components/modals/ViewStatsModal.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect, useState } from 'react'
+import { Contest, ContestStats } from '../../interfaces'
+import Modal from '@app/ui/components/Modal'
+import { Group, Label, LabelText, TextArea } from '@app/ui/components/Form'
+import { Button, StackContainer } from '@app/ui/components'
+import ContestApi from '../../api'
+import RankingApi from '@app/ranking/api'
+import { Ranking } from '@app/ranking/interfaces'
+import { formatLanguageName } from '@app/ranking/transform/format'
+
+function generateBlogPostSkeleton(stats: ContestStats, ranking: Ranking[]) {
+  const text = (textContent: string) => document.createTextNode(textContent)
+  const el = (
+    tagName: string,
+    textContent: string | number,
+    children: (HTMLElement | Text)[] = [],
+    attributes?: Record<string, string>,
+  ) => {
+    const el = document.createElement(tagName)
+    el.textContent = '' + textContent
+    for (const child of children) {
+      el.appendChild(child)
+    }
+    if (attributes) {
+      for (const attr in attributes) {
+        if (attributes.hasOwnProperty(attr)) {
+          el.setAttribute(attr, attributes[attr])
+        }
+      }
+    }
+    return el
+  }
+
+  const winner = ranking[0]
+
+  return el('div', '', [
```

Comment:
This I would prefer to be written in idiomatic React code. Seems like it's re-implementing what React already does for us? `renderToString(element)` can be used to get the raw html from a React component.

Ref. https://reactjs.org/docs/react-dom-server.html#rendertostring

Reply:
Done! Though I used `renderToStaticMarkup` which is the same thing but without any react attributes. https://reactjs.org/docs/react-dom-server.html#rendertostaticmarkup